### DISCUSSION TITLE
Importing class methods in strict mode

### DISCUSSION THREAD

**Discussion by excubo-jg** [microsoft/pyright#5862] [Q&A] [answered]
I am moving a repo which checks in basic mode without errors/warnings to strict mode. The code imports class methods into a file with the class definition (similar to https://stackoverflow.com/questions/1057934/importing-methods-for-a-python-class/1059377#1059377) which strict mode does not like. The following shows the gist of the code and the errors reported:

main_module.py:
```
class one_class:
	a_variable: bool

	from some_methods import method_a	# Import "method_a" is not accessed
						# Type of "method_a" is partially unknown
						#     Type of "method_a" is "(self: Unknown, a: int) -> None"
	from more_methods import method_b	# Import "method_b" is not accessed
						# Type of "method_b" is partially unknown
						#     Type of "method_b" is "(self: Unknown, b: int) -> None"

	def __init()__() -> None:
		pass
```

some_methods.py
```
def method_a(self, a: int) -> None	# Type of parameter "self" is unknown
	self.a_variable = False  		# Type of "a_variable" is unknown

```
more_methods.py
```
def method_b(self, b: int) -> None	# Type of parameter "self" is unknown
	self.method_a(0)		# Type of "method_a" is unknown

```

What would be the most efficient way to make the code amenable to strict checking?

**Comment by erictraut** [comment, marked as answer]
Wow, that's really gross code! If you're taking the time to make this code base type check in strict mode, then presumably you're open to removing such hackery. I would personally modify it to remove the class-scoped imports. Perhaps that's not possible or feasible in this case.

In any case, I agree that pyright shouldn't consider the symbol unaccessed in this case. I'll fix that.

As for the "partially unknown" error, I think that's legitimate. The type of the `self` parameter cannot be inferred in this case because the method is defined outside of the class. You will need to supply an explicit type annotation for `self` if you want to avoid the `Unknown` inferred type.

**Comment by excubo-jg** [comment]
Many thanks for your guidance.

I have not seen a tracking issue to fix the not accessed error. I am opening one to make sure this does fall through the cracks - if it is a dupe please remove.
